### ISSUE TITLE
Cinnamon Desktop slows after a few minutes, persists until restart

### ISSUE THREAD

**Issue by mckatzenelson** [linuxmint/muffin#402]
``Fedora 28, x64, Cinnamon desktop

After a few minutes from login to a Cinnamon desktop session, all of the windows become jumpy or slow, or unresponsive.  The problem is cleared by a restart using "Alt-F2 r", and then reappears after a few minutes.

Also, newly created windows blank to background when dragged to the second monitor.

**Comment by brownsr** [comment]
Worth checking what is in your .xsession-errors log.  If you have an error case that is causing excessive logging then that could slow things I suspect.  We'll need to see logs anyway in order to reproduce.

**Comment by jaszhix** [comment]
The windows disappearing was fixed in 4.0.3, and I haven't reproduced it since #382.

@mckatzenelson Are you seeing windows flicker black or disappear transparently? Some basic info is needed here, like the GPU, graphics driver, cinnamon version, etc. Please fill out the [issue template](https://github.com/linuxmint/muffin/blob/master/.github/ISSUE_TEMPLATE.md).

**Comment by mckatzenelson** [comment]
4.0.3 of what?

The window flickers to reveal the wallpaper,  back and forth in blocks and sometimes all of it.

.xsession-errors attached

[xsession-errors.txt](https://github.com/linuxmint/muffin/files/2710907/xsession-errors.txt)

Here are some lines from dmesg that might be of interest:

Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz (family: 0x6, model: 0x3a, stepping: 0x9)
 Linux version 4.19.10-200.fc28.x86_64
i915 0000:00:02.0: vgaarb: changed VGA decodes: olddecodes=io+mem,decodes=io+mem:owns=io+mem
[drm] Initialized i915 1.6.0 20180719 for 0000:00:02.0 on minor 0
ACPI: Video Device [GFX0] (multi-head: yes  rom: no  post: no)
i915 0000:00:02.0: fb0: inteldrmfb frame buffer device

**Comment by leigh123linux** [comment]
@mckatzenelson  posting the output for this command should provide the required info

```
inxi -GSM
```

**Comment by mckatzenelson** [comment]
Excellent, here it is:

System:    Host: desktop Kernel: 4.19.10-200.fc28.x86_64 x86_64 bits: 64 Desktop: Cinnamon 4.0.5 
           Distro: Fedora release 28 (Twenty Eight) 
Machine:   Type: Desktop Mobo: N/A model: N/A serial: <root required> BIOS: American Megatrends v: 4.6.5 date: 11/02/2012 
Graphics:  Device-1: Intel Xeon E3-1200 v2/3rd Gen Core processor Graphics driver: i915 v: kernel 
           Display: x11 server: Fedora Project X.org 1.19.6 driver: intel resolution: 1920x1200~60Hz, 1920x1080~60Hz 
           OpenGL: renderer: Mesa DRI Intel Ivybridge Desktop v: 4.2 Mesa 18.0.5

**Comment by jaszhix** [comment]
> 4.0.3 of what?

Muffin 4.0.3

**Comment by mckatzenelson** [comment]
The installed version is
       muffin.x86_64      4.0.4-1.fc28

**Comment by jaszhix** [comment]
Ok, you're missing the patch from #394. Is muffin 4.0.5 available?

**Comment by leigh123linux** [comment]
@jaszhix  yes
@mckatzenelson The updates in testing might fix this

https://bodhi.fedoraproject.org/updates/FEDORA-2018-af0019dd50

```
sudo
dnf --enablerepo=u*g update cinnamon muffin --refresh
```

**Comment by mckatzenelson** [comment]
If I install that, is it always going to get it from there when I update the system some months from now?
I hope to return to mainstream when that is integrated into the regular repositories

**Comment by mckatzenelson** [comment]
(I actually update more often than that)

**Comment by mckatzenelson** [comment]
okay, installing now

**Comment by mckatzenelson** [comment]
Nope!  Not fixed.   A window dragged to the second monitor, still flickers.

How do I back out of the update from testing?

**Comment by mckatzenelson** [comment]
And the response is still intermittent and becomes jumpy.

**Comment by leigh123linux** [comment]
@mckatzenelson  https://bodhi.fedoraproject.org/updates/FEDORA-2018-af0019dd50 is queued for stable repo.

If you wish to downgrade run

```
sudo dnf  downgrade cinnamon muffin
```

**Comment by jaszhix** [comment]
@mckatzenelson Does this occur when the desktop is shown only on the second monitor? If so, if you maximize a window on that monitor, then move a window to it, does it flicker?

**Comment by mckatzenelson** [comment]
I am not sure what that means.

Both monitors are active,  the windows typically open on the primary monitor (DVI) and the problem happens when I drag a window to the secondary (VGA).

**Comment by jaszhix** [comment]
I mean, do you see a difference in the flickering behavior if the desktop wallpaper is visible on the secondary monitor when a window is dragged to it. When I did see windows blink in the way I think you're describing, it only occurred when no other windows were open and the desktop was shown.

This might suggest the problem may be in how the background actor is being handled.

**Comment by mckatzenelson** [comment]
I'm not sure how to turn off the wallpaper rather than replacement.   The controls seems to only provide options for selecting another wallpaper.

Another clue might be that the worst offenders seem to be firefox and thunderbird.

Since the update however, I have not had to restart the desktop.

**Comment by mckatzenelson** [comment]
New information on blanking!

I start a program that opens a new window,  and wherever I pass the mouse over it, it blanks to background, until I click somewhere outside of that window.

I have observed this in VLC and the mozilla programs.  They are the only ones I use very often, that remember which window they were last opened it.

**Comment by mckatzenelson** [comment]
The performance problem has become unbearable.  Within a few minutes of a fresh boot, the desktop responds in fits and starts.  Its really unusable. I think the severity should be elevated.

**Comment by AdmiralAsshat** [comment]
Experiencing similar performance issues as well.

```
user@malebolge text]$ inxi -GSM
System:    Host: malebolge Kernel: 5.2.9-200.fc30.x86_64 x86_64 bits: 64 Desktop: Cinnamon 4.2.3 
           Distro: Fedora release 30 (Thirty) 
Machine:   Type: Laptop System: Dell product: XPS 13 9343 v: N/A serial: <root required> 
           Mobo: Dell model: 0310JH v: A00 serial: <root required> UEFI: Dell v: A18 date: 09/27/2018 
Graphics:  Device-1: Intel HD Graphics 5500 driver: i915 v: kernel 
           Display: x11 server: Fedora Project X.org 1.20.5 driver: modesetting unloaded: fbdev,vesa 
           resolution: 1920x1200~60Hz 
           OpenGL: renderer: Mesa DRI Intel HD Graphics 5500 (Broadwell GT2) v: 4.5 Mesa 19.1.5 
```

Let me know if there's anything else I can provide to help get some eyeballs on this issue.

**Comment by mckatzenelson** [comment]
The bug persists.     Really folks, the cinnamon desktop is close to un-usable because of this.

Current setup: 

Fedora 30
cinnamon.x86_64  4.2.4-2.fc30
muffin.x86_64  4.2.2-1.fc30
Dual monitor system, x64, intel i7 and motherboard, using the built-in video hardware.

After a few minutes from login, all of the windows become jumpy or slow, or unresponsive.  The problem is cleared by a restart of the desktop, and then reappears after a few minutes.

**Comment by AndreiTS** [comment]
Just encounter this problem in mint 22. Restarting cinnamon didn't fix it, had to restart everything. I was facing timeouts when using apt and flatpak and opening programs was taking several seconds.

System:
  Host: andrei-pc Kernel: 6.8.0-55-generic arch: x86_64 bits: 64
  Desktop: Cinnamon v: 6.4.8 Distro: Linux Mint 22.1 Xia
Machine:
  Type: Desktop System: Gigabyte product: N/A v: N/A
    serial: <superuser required>
  Mobo: Gigabyte model: B75M-D3H serial: <superuser required>
    UEFI: American Megatrends v: F12 date: 10/26/2012
Graphics:
  Device-1: NVIDIA GP107 [GeForce GTX 1050] driver: nvidia v: 550.120
  Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.6 driver: X:
    loaded: nouveau unloaded: fbdev,modesetting,vesa failed: nvidia
    gpu: nvidia,nvidia-nvswitch resolution: 1: 1920x1080~60Hz
    2: 1920x1080~60Hz
  API: EGL v: 1.5 drivers: nvidia,swrast platforms: x11,surfaceless,device
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: nvidia mesa v: 550.120
    renderer: NVIDIA GeForce GTX 1050/PCIe/SSE2
